BUILD.gn

```diff
@@ -1362,7 +1367,12 @@ if (spvtools_build_executables) {
   }
 
   executable("spirv-link") {
-    sources = [ "tools/link/linker.cpp" ]
+    sources = [ 
```

Comment:
this target depends on :`spvtools_link` already, so don't add the sources here.

Also, this removes tools/link/linker.cpp which provides `main`